### PR TITLE
ODC-5863-github templates

### PR DESCRIPTION
**Pull request by makambalaji**
**Description**:
Github templates for Design Gherkin scripts and Automation pr's

**Jira Id:** https://issues.redhat.com/browse/ODC-5863

### CODE REVIEW

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
It's good, but my main leading comment on responsibilities is still not defined.

Author has obligations that are pretty obvious; make sure the PR makes sense, touches the right files, works, etc

But reviewers come in two types as of today's standards. QE Peer Review and Dev Validation Review. 

- QE Peer Review checks for "Gherkin" like script states and helps validate everything.
- Dev Validation Review is more (at least from my understanding) a check against scenarios to see if something was missed or not given a test

These templates help leaps and bounds over not having them imo. But they don't detail out responsibilities -- not sure that's really what a template should do, but it has value to make sure these are conveyed properly somewhere (if not in these templates).

We need to avoid situations where a test "looks good" but never would work when put to the test. Not saying this PR/templates need to deal with that, but that's the core of my original issue.

File: frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+**Screen shots**:
+<!--   -->
+
+**Browser conformance**:
+<!-- To mark tested browsers, use [x] -->
+- [ ] Chrome
+- [ ] Firefox
+- [ ] Safari
+- [ ] Edge
```
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md:44]: Maybe in future, we can plan. But as now only chrome & firefox

File: frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+**Pre-conditions/setup:**
+<!-- If any setup required while performing epic validation [which is not mentioned in Back ground section], mention the details -->
+
+**Checks for approving Epic scenarios Automation PR:**
+<!-- Below criteria should met before approving the pr, use [x] -->
+- [ ] Execute the @to-do tagged gherkin scripts manually
+- [ ] Convert the @to-do gherkin scripts to cypress automation scripts
+- [ ] Once scripts are automated, replace tag @to-do with @epic-number
+- [ ] Execute the scripts in Remote cluster
```
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md:16]: This helps to make sure the author is not missing any steps in the automation scripting process.

File: frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+<!-- Briefly describe Acceptance criteria of Epic -->
+
+**Pre-conditions for Epic validation:**
+<!-- If any setup required while performing epic validation [which is not mentioned in Back ground section], mention the details -->
+
+**Checks required for approving PR:**
+<!-- Below criteria should met before approving the pr, use [x] -->
+- [ ] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
+- [ ] Add @to-do for automation possible scenarios
+- [ ] Add @regression or @smoke based on the importance of the scenario
```
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md:15]: Good point, I will include "@un-verified for the test scenarios not verified by ODC QE". Going forward other operators teams are also contributing test scenarios to our git repository

File: frontend/.gherkin-lintrc [hunk 4]
```diff
@@ -12,8 +12,8 @@
   "no-empty-file": "on",
   "no-scenario-outlines-without-examples": "on",
   "name-length": "off",
-  "allowed-tags": ["on", {"tags": ["@manual", "@regression", "@smoke", "@add-flow", "@topology", "@knative", "@guided-tour", "@gitops", "@helm", "@monitoring", "@pipelines", "@virtualization", "@web-terminal", "@vulnerability", "@knative-eventing", "@knative-camelK", "@kafka"]}],
-  "no-restricted-tags": ["on", {"tags": ["@watch", "@wip", "@debug", "@to-do"]}],
+  "allowed-tags": ["on", {"tags": ["@e2e", @codeready-workspaces, "@health-checks", "@service-mesh",  "@manual", "@regression", "@smoke", "@add-flow", "@topology", "@knative", "@guided-tour", "@gitops", "@helm", "@monitoring", "@pipelines", "@virtualization", "@web-terminal", "@vulnerability", "@knative-eventing", "@knative-camelK", "@kafka", "@to-do", "un-verified"]}],
+  "no-restricted-tags": ["on", {"tags": ["@watch", "@wip", "@debug"]}],
```
Comment (andrewballantyne) [inline frontend/.gherkin-lintrc:16]: I'm confused, this doesn't appear related to the templates we are making 🤔

File: frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md:16]: Still confusing on what this means, but if you think it drives value to the QE team, might as well keep it 🙂 

I'm not sure how to execute it without a remote cluster, but it's useful as a checkbox I suppose 😄

File: frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md:44]: That's fair... not sure we need to jump the gun today, but onwards and upwards, always looking to the future 🙂

File: frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md:15]: It's not so much not verified by us... it's not ever tested. If you write a script and you don't execute it, that's a problem. The two scenarios I can see are:

1. It currently can work on a cluster today, and it was not tested because of the "Bystander Effect" and everyone assumed someone else tested it
2. It cannot be executed on the cluster because we are doing it in parallel with the Dev work and the feature is not merged yet

If it's 1, this is the problem we are having and I would like that gap to be closed. Any author of these Gherkin scripts, be it our team or not, should be executing them and thus should _never_ have the `@un-verified` annotation (within' reason).

If it's 2, then that's a good reason for us to use this label. That way we can swing back when doing epic verification or once at the end of the sprint to test all of them and remove the `@un-verified` annotation.

I hope this is clear as it's the foundation for all this effort. If you have any questions or concerns, please let me know.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/.gherkin-lintrc (hunk 4 above)
Comment (makambalaji) [reply frontend/.gherkin-lintrc:16]: Added @un-verified tag and included other tags as well

File: frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md:15]: That makes sense

File: frontend/.gherkin-lintrc (hunk 4 above)
Comment (andrewballantyne) [reply frontend/.gherkin-lintrc:16]: Well, two things then.

1. Why are you taking this opportunity to just get extra stuff into the PR? PRs should have goals and not become just a dumping ground for any changes -- I think your non-template like changes here is not ideal (I suspect you're doing it out of convenience)
2. Shouldn't it be `@un-verified`? Today you have:
>  "@to-do", "un-verified"

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/.github/PULL_REQUEST_TEMPLATE/qe-automation-pr.md:16]: Earlier We use to do execution on localhost. So I want to avoid that in future

File: frontend/.gherkin-lintrc (hunk 4 above)
Comment (makambalaji) [reply frontend/.gherkin-lintrc]: Deleted the file, I can raise a seperate pr

Comment (andrewballantyne) [reply frontend/.gherkin-lintrc]: I'm so confused... I asked why are you doing things that have nothing to do with the PR, so you respond with "I'll undo everything"? Help me understand here what I am missing... Do you not need to do `"@to-do", "@un-verified"` part of this work to support the work we are talking about in this PR?

> `"@e2e", @codeready-workspaces, "@health-checks", "@service-mesh",`

Has nothing to do with the discussion and work we are trying to do here, or am I mistaken? 

There is no reason to over reach and clean up things that are not related (ie tech debt), but there is also no reason not to address the topic at hand.

**Review by andrewballantyne — APPROVED** [review]
Since this work is not related to anything code related I'm just going to approve and get this merged. This was started by a lack of proper due-process in writing of Gherkin Scripts and I feel that has been sorta covered by the templates. It's better than nothing but we don't need to take a week to merge this.

/lgtm

File: frontend/.gherkin-lintrc (hunk 4 above)
Comment (makambalaji) [reply frontend/.gherkin-lintrc]: I understand your point, That's the reason, I removed the file and raised a separate PR to fix .gherkin-lintrc file
